lib/queues.js

```diff
@@ -260,16 +270,64 @@ class Queue {
   runOnce() {
     assert(!this._processing, 'Cannot call once while continuously processing jobs');
     const hasHandlers = (this._handlers.size > 0);
+
     if (hasHandlers) {
 
       debug('Waiting for jobs on queue %s', this.name);
-      return this._getClientPromise.then(client => this._reserveAndProcess(client));
+
+      if (this._delayedJobs.size > 0) {
+
+        const jobIDs = this._getDelayedJobIDs();
+
+        return this._kickJobs(jobIDs)
+          .then(() => this._getClientPromise)
+          .then(client => this._reserveAndProcess(client));
+
+      } else {
+
+        return this._getClientPromise
+          .then(client => this._reserveAndProcess(client));
+
+      }
 
     } else
       return Promise.resolve(false);
   }
 
 
+  // Returns a list of stored delayed jobs IDs that are ready to be processed
+  _getDelayedJobIDs() {
+    const jobIDs = [];
+
+    this._delayedJobs.forEach(function(readyTime, id) {
+      if (readyTime <= Date.now())
+        jobIDs.push(id);
+    });
+
+    return jobIDs;
+  }
+
+
+  // Moves delayed jobs to the ready queue (to be processed) in test mode
+  _kickJobs(jobIDs) {
+    const self = this;
+
+    return self._putClientPromise
+      .then(function(client) {
+
+        return Bluebird.map(jobIDs, function(id) {
+          return client.kickJob(id)
+            .then(() => self._delayedJobs.delete(id))
+            .catch(function(error) {
+              if (error.message === 'NOT_FOUND')
+                debug(`Could not find and delete job with id: ${id}`);
+              else
+                throw new Error(`Error deleting job with id: ${id} with message: ${error.message}`);
```

Comment:
Ah. I added this because after kicking a job that was already processed, there were `NOT_FOUND` error in this test: https://github.com/assaf/ironium/blob/master/test/queue/with_delay_test.js#L48

Another solution I looked into: write a peek by id method to check if the job has been processed or not before issuing a kick command.